### PR TITLE
Add license and distribution license generation

### PR DESCRIPTION
**Pull request by Delerme**
- Add a license to the library
- Add a distribution license generation step to build (which includes licenses
  of direct production dependencies)

TODO:
- Get final license from Legal
- Update package.json with SPDX license string
- Include package.json in dist on build

### CODE REVIEW

File: conf/gulp-tasks/library.gulpfile.js [hunk 1]
```diff
@@ -113,19 +116,72 @@ function compileCSS () {
...
+          const license = await licensePromise;
+          return [allLicenses, `${license.name} License`, license.text].join('\n\n');
+        }, Promise.resolve(''))
+        .catch(err => cb(err, file));
+
+      file.contents = Buffer.concat([file.contents, Buffer.from(allDepLicenses)]);
+      cb(null, file);
+    });
+  });
+}
```
Comment (Delerme) [inline conf/gulp-tasks/library.gulpfile.js]: I think I can make this cleaner but wanted to open this up for feedback early

File: conf/gulp-tasks/library.gulpfile.js [hunk 2]
```diff
@@ -113,19 +116,72 @@ function compileCSS () {
...
+    omitVersion: true
+  };
+
+  return through.obj((file, _, cb) => {
+    licenseCrawler.dumpLicenses(licenseCrawlerOptions, async (err, deps) => {
+      if (err) {
+        cb(err, null);
+      }
+
+      let allDepLicenses = await Object.keys(deps)
```
Comment (data-enabler) [inline conf/gulp-tasks/library.gulpfile.js:152]: I'd suggest `Object.entries()`

File: conf/gulp-tasks/library.gulpfile.js [hunk 3]
```diff
@@ -113,19 +116,72 @@ function compileCSS () {
...
+          if (!licenseTextRes.ok) {
+            cb(text, file);
+          }
+
+          return {
+            ...license,
+            text: text.trim()
+          };
+        })
+        .reduce(async (allLicensesPromise, licensePromise) => {
```
Comment (data-enabler) [inline conf/gulp-tasks/library.gulpfile.js:169]: I think you probably want to use `Promise.all()` here

**Comment by benmcginnis** [comment]
@data-enabler what was the process y'all went through to get the license from legal? I'm curious do so for some of our tools.

**Comment by data-enabler** [comment]
> @data-enabler what was the process y'all went through to get the license from legal? I'm curious do so for some of our tools.

No idea, unfortunately @benmcginnis. I wasn't really involved in that part of the process.

**Pull request closed without merging**